File: frontend/rfindustriescore/src/lib/beans/core/ResponseMethod.ts
import { ResponseStatus } from "../../constants/core/ResponseStatus";
import { Message } from "./Message";

export class ResponseMethod<T> {
  data?: T;
  status: ResponseStatus = ResponseStatus.SUCCESS;
  error?: Error;
  messages: { [key: string]: Message[] } = {};
  httpStatus: number = 0;
}

File: backend/java/rf-core/src/main/java/com/rfindustries/core/beans/ResponseMethod.java
package com.rfindustries.core.beans;

import com.rfindustries.core.constansts.MessageLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResponseMethod<T> {
    private T data;
    private Map<MessageLevel, List<Message>> messages;
}
